libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -525,10 +586,29 @@ void AP_Baro::update(void)
         sensors[i].healthy = (now - sensors[i].last_update_ms < 500) && !is_zero(sensors[i].pressure);
     }
 
+    // find current GPS altitude
+    float gps_alt = 0.0f;
+    bool process_gps_alt = false;
+    if (!is_zero(_gps_adj_step) && _gps_calibrated) {
+        float gps_vacc;
+        if (_gps->vertical_accuracy(gps_vacc)) {
```

Comment:
They are called a lot of times though.
Extra parameters - it will take time and stack and also more code and changes from calling side. Is it really worth it?
What is bad with such dependency? We have it in many places.